src/fantasi/Tuura/Fantasi/Main.hs

```diff
@@ -14,13 +17,27 @@ main = do
     let graphMLPath           = optGraphML options
         graphVHDLPath         = optGraphName options
         simulationEnvVhdlPath = optSimName options
+        runAlias              = optAliasHub options
+
+    when runAlias (print "Running Alias")
 
     -- parse graph
-    pangraph <- ((maybe (error "file or graph is malformed") id) . P.parse) <$> readFile graphMLPath
+    maybePangraph <- (`getPangraph` options) <$> readFile graphMLPath
+    let pangraph = fromMaybe (error "Pangraph is nothing!") maybePangraph 
+    -- let Just pangraph =  maybePangraph 
+    
     let graphVHDL   = VHDL.writeGraph pangraph
     let simEnvVHDL  = VHDL.writeEnvironment pangraph
 
     -- output vhdl graph
     writeFile graphVHDLPath graphVHDL
     -- output vhdl simulation environment
     writeFile simulationEnvVhdlPath simEnvVHDL
+
+getPangraph :: ByteString -> Options -> Maybe Pangraph
```

Comment:
It feels strange to intertwine parsing and aliasing logic. I suggest you move this to the `main` function: first parse a graph, reporting parse errors, and then perform your transformation.